### DISCUSSION TITLE
Adding dependencies to package.json

### DISCUSSION THREAD

**Discussion by nskmda** [node-gradle/gradle-node-plugin#250] [Q&A] [answered]
What's the correct way to add dependencies to `package.json` via this plugin?
How can I configure it to actually run `npm install <module name>`?

When I'm trying to define an NPM task like this

```groovy
task installPackages(type: NpmTask) {
  args = ['install', 'webpack']
}
```

I get an error like the following:

```
> Process 'command 'C:\Projects\react\.gradle\nodejs\node-v16.17.1-win-x64\npm.cmd'' finished with non-zero exit value 1
```

I'm running Windows 10, Gradle 7.5.1, Amazon JDK 11 with the latest (currently) version of the plugin.

I know I can manually add all those dependencies to `package.json` but I wanted to see if I could have them listed somewhere instead.
So that all the proper stuff (like version) simply gets updated whenever I run my Gradle task.

Can you please let me know what I'm doing wrong?

**Comment by nskmda** [comment]
UPDATE: i guess there's something conceptually wrong in what i'm doing.

I have 

```
      "dev:server": "nodemon --exec babel-node src/server/server.js --ignore dist/"
```

in `package.json`

```
task devBundler(type:NpmTask)
{
    args = ['run', 'dev:server']
}
```
fails with the same error code and message.

**Comment by deepy** [comment, marked as answer]
The general setup would be something like [examples/simple-node/npm](https://github.com/node-gradle/gradle-node-plugin/tree/3.4.0/examples/simple-node/npm), where you handle `package.json` through manual interactions with a local `npm` (or with an editor if you're particularly brave)

The commands look correct to me based on the documentation, but try with `npmCommand` set to `run` and `dev:server` as `args` instead.
There may be a documentation issue here, I need to go through the documentation properly and clean it up.

**Reply by nskmda** [reply]
Perfecto! Worked as expected. Thanks a lot!
